NDTensors/src/zeros/zeros.jl

```diff
@@ -0,0 +1,45 @@
+struct Zeros{ElT,N,Axes,Alloc<:AbstractArray} <: AbstractArray{ElT,N}
+  z::FillArrays.Zeros
+  function NDTensors.Zeros{ElT,N,Alloc}(inds::Tuple) where {ElT,N,Alloc}
+    @assert eltype(Alloc) == ElT
```

Comment:
This would get constrained automatically if you write the type definition as:
```julia
struct Zeros{ElT,N,Axes,Alloc<:AbstractArray{ElT,N}} <: AbstractArray{ElT,N}
```

Reply:
I guess I am wondering if `Alloc` needs to have `ElT` and `N` since its functionally just a place holder?  (this follows from the thread before) We could just store `Zero{Float32, 20, (ind_1, ind_2, ..., ind_20), Vector}` and then process the `Alloc` properly when its actually instantiated later?

Reply:
I agree that in principle we could design it that way, but it seems better to me to store it in a way where it can be directly used to allocate an array instead of having to process the type first before being able to use it. For example, then the `Alloc` type could only be used properly by some accessor function that filled in the type parameters on the fly (say `allocated_type(::NDTensors.Zeros)` or something like that) but if you access the type from the type parameter directly it wouldn't be complete. Seems better to process the type properly right off the bat.

Something we could do so that it isn't too annoying to construct a `NDTensors.Zeros` with a custom allocated array type is have a constructor:
```julia
function Zeros(alloc::Type{<:AbstractArray}, dims...)
  # 
end

function Zeros{ElT}(alloc::Type{<:AbstractArray}, dims...) where {ElT}
  # 
end
```
which could process the allocated type `alloc` to fill in missing parameters based on `dims` and possibly a default element type if the element type of `alloc` or `Zeros` isn't specified. That way you could be a bit looser about how you define it to begin with, and use `Zeros(Array, 2, 2)`, `Zeros{Float64}(Array, 2, 2)`, `Zeros(Array{Float64}, 2, 2)`, etc. That would be helpful for GPU types like `CuArray` which have the extra buffer type.